cura/Settings/MachineManager.py

```diff
@@ -1496,3 +783,268 @@ def setSettingForAllExtruders(self, setting_name: str, property_name: str, prope
         for key, extruder in self._global_container_stack.extruders.items():
             container = extruder.userChanges
             container.setProperty(setting_name, property_name, property_value)
+
+    #
+    # New
+    #
+
+    # We not fetch it from _current_root_material_id, but later we can get it from somewhere else
+    @pyqtProperty("QVariantList", notify = rootMaterialChanged)
+    def currentExtruderPositions(self):
+        return sorted(list(self._current_root_material_id.keys()))
+
+    @pyqtProperty("QVariant", notify = rootMaterialChanged)
```

Comment:
It seems like we're depending on QML to update the cached `self._current_root_material_id` here. If the currentRootMaterialId is not used in QML, then the `_current_root_material_id` field will not get updated. This is unreliable.

Reply:
We will try to fix this later.

Reply:
Fixed in a87db2d721e7779c4b6d3650aa60155e0de22ea4.